client/common/client.c

```diff
@@ -208,6 +208,46 @@ static BOOL freerdp_client_settings_post_process(rdpSettings* settings)
 		settings->Fullscreen = TRUE;
 	}
 
+	/* deal with the smartcard / smartcard logon stuff */
+	if (settings->SmartcardEmulation)
+	{
+		/* if no pin is defined on the smartcard emulation use the user password */
+		if (!settings->SmartcardPin)
+		{
+			if (!settings->Password)
+			{
+				WLog_ERR(TAG, "No pin or password defined for smartcard emu");
+				goto out_error;
+			}
+
+			if (!freerdp_settings_set_string(settings, FreeRDP_SmartcardPin, settings->Password))
+			{
+				WLog_ERR(TAG, "error when setting smartcard pin to user password");
+				goto out_error;
+			}
+		}
+	}
+
+	if (settings->SmartcardLogon)
+	{
+		settings->TlsSecurity = TRUE;
+		settings->RedirectSmartCards = TRUE;
+		settings->DeviceRedirection = TRUE;
+		freerdp_settings_set_bool(settings, FreeRDP_PasswordIsSmartcardPin, TRUE);
+
+		if (!settings->Password && settings->SmartcardEmulation)
```

Comment:
After checking what `SmartcardEmulation` actually is, can´t you just remove it? I don´t see how this has anything to do with emulation but just code duplication